### PR TITLE
(themes) Add `tokyo-night-dark`

### PR DESCRIPTION
**Pull request by Vanderscycle**
Added a color-close tokyonight theme

Added the tokyonight theme inspired by [vscode](https://github.com/enkia/tokyo-night-vscode-theme)

### Changes

new css file supporting tokyonight

### Checklist
- [x] Added markup tests (todo)
- [x] Updated the changelog at `CHANGES.md`

### CODE REVIEW

**Comment by joshgoebel** [comment]
Please show us what it looks like and attach the results of running our theme checker utility against your theme.

**Comment by Vanderscycle** [comment]
@joshgoebel,
Hey thank you for getting back to me here's a snapshot of what the code looks like. Its as accurate as I can do right now with the new code.  

image

I am using svelte-highlight which in turns uses highlight.js for themes. I am not familiar with the code base and how to test my theme the hightlight.js  [way](https://highlightjs.readthedocs.io/en/latest/building-testing.html). So I am using svelte highlight with my custom file.

```svelte
<style lang="postcss"></style>

<script lang="ts">


$: code = `  // swagger config
  const config = new DocumentBuilder()
    .setTitle('A51-21e8 node')
    .setDescription(
      'Still confused about levelDown and its nestJs integrations',
    )
    .setVersion('.01')
    .build();
  const document = SwaggerModule.createDocument(app, config);
  SwaggerModule.setup('/docs', app, document);

  await app.listen(process.env.SERVER_PORT ?? 3100, '0.0.0.0', async (err, address) => {
    if (err) {
      Logger.warn(err)
      process.exit(1)
    } else {
      Logger.log("Application is running" , 'main.ts')
    }
  })
`;
import { Highlight } from "svelte-highlight";
import typescript from "svelte-highlight/src/languages/typescript";
import tokyoNightDark from "../lib/ui/tokyo-night-dark.css";
</script>

<svelte:head>
  {@html tokyoNightDark}
</svelte:head>

<template>
  <div class="bg-terminalStorm"> <!-- terminalStorm: "#24283b" -->
    <pre><code><Highlight language="{typescript}" code="{code}" /></code></pre>
  </div>
</template>
```

**Comment by joshgoebel** [comment]
Looks pretty.  Please run `./tools/checkTheme.js` against your CSS file and paste results here...

**Comment by Vanderscycle** [comment]
```
❯  ./tools/checkTheme.js src/styles/tokyonight-dark.css
Templates/HTML/XML, etc.
- Theme does not fully support.
- scope attr is not highlighted
  (css: .hljs-attr)
- scope template-tag is not highlighted
  (css: .hljs-template-tag)

CSS/Less/etc
- Theme does not fully support.
- scope selector-attr is not highlighted
  (css: .hljs-selector-attr)
- scope selector-pseudo is not highlighted
  (css: .hljs-selector-pseudo)

Markup (Markdown, etc)
- Theme does not fully support.
- scope code is not highlighted
  (css: .hljs-code)
- scope formula is not highlighted
  (css: .hljs-formula)

program code
- Theme does not fully support.
- scope property is not highlighted
  (css: .hljs-property)
- scope subst is not highlighted
  (css: .hljs-subst)
- scope doctag is not highlighted
  (css: .hljs-doctag)
- scope attr is not highlighted
  (css: .hljs-attr)

nice to haves (optional, but many grammars use)
- Theme does not fully support.
- scope meta keyword is not highlighted
  (css: .hljs-meta .hljs-keyword)
- scope meta string is not highlighted
  (css: .hljs-meta .hljs-string)

high fidelity highlighting (this is optional)
- Theme does not fully support.
- scope title.class is not highlighted
  (css: .hljs-title.class_)
- scope title.class.inherited is not highlighted
  (css: .hljs-title.class_.inherited__)
- scope punctuation is not highlighted
  (css: .hljs-punctuation)
- scope operator is not highlighted
  (css: .hljs-operator)
- scope title.function is not highlighted
  (css: .hljs-title.function_)
- scope char.escape is not highlighted
  (css: .hljs-char.escape_)
- scope variable.language is not highlighted
  (css: .hljs-variable.language_)

Accessibility Report
┌───────┬────────────────────────────────────────┬──────────┬──────────┐
│ ratio │ selector                               │ fg       │ bg       │
│ 2.35  │ .hljs-comment,.hljs-quote              │ #565f89  │ #24283b  │
│ 5.51  │ .hljs-variable,.hljs-template-variabl… │ #f7768e  │ #24283b  │
│ 7.16  │ .hljs-number,.hljs-built_in,.hljs-lit… │ #ff9e64  │ #24283b  │
│ 7.28  │ .hljs-attribute                        │ #e0af68  │ #24283b  │
│ 7.97  │ .hljs-string,.hljs-symbol,.hljs-bulle… │ #9ece6a  │ #24283b  │
│ 5.78  │ .hljs-title,.hljs-section              │ #7aa2f7  │ #24283b  │
│ 6.3   │ .hljs-keyword,.hljs-selector-tag       │ #bb9af7  │ #24283b  │
│ 14.57 │ .hljs                                  │ #ffffff  │ #24283b  │
└───────┴────────────────────────────────────────┴──────────┴──────────┘

```

**Comment by joshgoebel** [comment]
> Does that mean that I am missing several css tags like .hjls-doctag/hljs-code?

Exactly. :-)  Generally you don't want to see "- Theme does not fully support." messages.

Also, I'd recommend pulling the contrast a bit on your ` .hljs-comment,.hljs-quote` colors... we really don't like to see anything below a 3... and 4.5 is what it would take to fully be WCAG 2.0 Level AA compliant.

Reference: https://webaim.org/resources/contrastchecker/

**Comment by Vanderscycle** [comment]
@joshgoebel, made the changes. Is there a way to quickly generate a preview of my theme?
```
❯ ./tools/checkTheme.js src/styles/tokyonight-dark.css
nice to haves (optional, but many grammars use)
- Theme does not fully support.
- scope meta keyword is not highlighted
  (css: .hljs-meta .hljs-keyword)
- scope meta string is not highlighted
  (css: .hljs-meta .hljs-string)

high fidelity highlighting (this is optional)
- Theme does not fully support.
- scope punctuation is not highlighted
  (css: .hljs-punctuation)
- scope title.function is not highlighted
  (css: .hljs-title.function_)
- scope char.escape is not highlighted
  (css: .hljs-char.escape_)
- scope variable.language is not highlighted
  (css: .hljs-variable.language_)

Accessibility Report
┌───────┬────────────────────────────────────────┬──────────┬──────────┐
│ ratio │ selector                               │ fg       │ bg       │
│ 2.76  │ .hljs-comment                          │ #565f89  │ #1a1b26  │
│ 6.46  │ .hljs-variable,.hljs-template-variabl… │ #f7768e  │ #1a1b26  │
│ 8.4   │ .hljs-number,.hljs-built_in,.hljs-lit… │ #ff9e64  │ #1a1b26  │
│ 8.55  │ .hljs-attribute                        │ #e0af68  │ #1a1b26  │
│ 9.96  │ .hljs-title,.hljs-title.class_,.hljs-… │ #7dcfff  │ #1a1b26  │
│ 9.35  │ .hljs-quote,.hljs-string,.hljs-symbol… │ #9ece6a  │ #1a1b26  │
│ 6.79  │ .hljs-code,.hljs-formula,.hljs-section │ #7aa2f7  │ #1a1b26  │
│ 7.39  │ .hljs-meta,.hljs-keyword,.hljs-operat… │ #bb9af7  │ #1a1b26  │
│ 10.59 │ .hljs                                  │ #c0caf5  │ #1a1b26  │
└───────┴────────────────────────────────────────┴──────────┴──────────┘
```

**Comment by joshgoebel** [comment]
>  Is there a way to quickly generate a preview of my theme?

From the source code you can build the `browser` target then open `build/demo/index.html` to browse all the themes.

**Comment by Vanderscycle** [comment]
Would you be able to point to the right variable names?

The browser env
image

The [TokyioNight](https://github.com/folke/tokyonight.nvim) Dark theme that I use on nvim 
image

**Comment by joshgoebel** [comment]
Use `tools/developer.html` for more visibility of what the highlighter is doing. It only requires you to first build the browser build.

**Comment by Vanderscycle** [comment]
ok so I played with the tools and it helped, these are my final results. 

```❯ ./tools/checkTheme.js src/styles/tokyonight-dark.css
nice to haves (optional, but many grammars use)
- Theme does not fully support.
- scope meta keyword is not highlighted
  (css: .hljs-meta .hljs-keyword)
- scope meta string is not highlighted
  (css: .hljs-meta .hljs-string)

Accessibility Report
┌───────┬────────────────────────────────────────┬──────────┬──────────┐
│ ratio │ selector                               │ fg       │ bg       │
│ 2.76  │ .hljs-meta,.hljs-comment               │ #565f89  │ #1a1b26  │
│ 6.46  │ .hljs-tag,.hljs-doctag,.hljs-selector… │ #f7768e  │ #1a1b26  │
│ 8.4   │ .hljs-variable,.hljs-template-variabl… │ #ff9e64  │ #1a1b26  │
│ 8.55  │ .hljs-built_in,.hljs-attribute         │ #e0af68  │ #1a1b26  │
│ 8.11  │ .hljs-selector-tag                     │ #2ac3de  │ #1a1b26  │
│ 9.96  │ .hljs-keyword,.hljs-title.function_,.… │ #7dcfff  │ #1a1b26  │
│ 10.26 │ .hljs-selector-tag                     │ #73daca  │ #1a1b26  │
│ 9.35  │ .hljs-quote,.hljs-string,.hljs-symbol… │ #9ece6a  │ #1a1b26  │
│ 6.79  │ .hljs-code,.hljs-formula,.hljs-section │ #7aa2f7  │ #1a1b26  │
│ 7.39  │ .hljs-name,.hljs-keyword,.hljs-operat… │ #bb9af7  │ #1a1b26  │
│ 10.59 │ .hljs-punctuation                      │ #c0caf5  │ #1a1b26  │
│ 7.04  │ .hljs                                  │ #9aa5ce  │ #1a1b26  │
└───────┴────────────────────────────────────────┴──────────┴──────────┘
```

File: CHANGES.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 ## Version 11.4.1
 
+Themes:
+- `tokyionight-dark` has been added and feedback/changes are welcome
```
Comment (joshgoebel) [inline CHANGES.md:4]:
```suggestion
- Added `TokyoNight-dark` theme [Henri Vandersleyen][]
```

File: src/styles/tokyonight-dark.css [hunk 2]
```diff
@@ -0,0 +1,112 @@
+/* Tokyo Night Blue Theme */
+/* Original theme https://github.com/enkia/tokyo-night-vscode-theme*/
```
Comment (joshgoebel) [inline src/styles/tokyonight-dark.css:2]: Please update this header using the format found in `default.css`.  Otherwise, looks good!

**Review by joshgoebel — CHANGES_REQUESTED** [review]

File: src/styles/tokyonight-dark.css (hunk 2 above)
Comment (Vanderscycle) [reply src/styles/tokyonight-dark.css]: done

File: src/styles/tokyonight-dark.css [hunk 3]
```diff
@@ -0,0 +1,120 @@
+/*!
+  Theme: Tokyionight-Dark
+  origin: https://github.com/enkia/tokyo-night-vscode-theme
+  Description: Original highlight.js style
+  Author: (c) Henri Vandersleyen <hvandersleyen@gmail.com>
+  Maintainer: @highlightjs/core-team
+  Website: https://highlightjs.org/
```
Comment (joshgoebel) [inline src/styles/tokyonight-dark.css:7]:
```suggestion
```

We do not automatically become tagged maintainer of all added themes.

**Comment by joshgoebel** [comment]
Sorry, one last thing.... should we not name this `tokyo-night-dark` for readability?

File: CHANGES.md [hunk 4]
```diff
@@ -1,5 +1,8 @@
 ## Version 11.4.1
 
+Themes:
+- Added `TokyoNight-dark` theme [Henri Vandersleyen][]
```
Comment (joshgoebel) [inline CHANGES.md:4]:
```suggestion
- Added `Tokyo-Night-dark` theme [Henri Vandersleyen][]
```

File: tools/developer.html [hunk 5]
```diff
@@ -71,6 +71,7 @@ <h3>Code</h3>
         Language: <select class="languages"><option value="">(Auto)</option></select>
         Theme: <select class="theme">
           <option>stackoverflow-light.css</option>
+          <option>tokyonight-dark.css</option>
```
Comment (joshgoebel) [inline tools/developer.html:74]:
```suggestion
          <option>tokyo-night-dark.css</option>
```

**Comment by joshgoebel** [comment]
@Vanderscycle Thanks fo much!

**Pull request merged**